modules/distribution/resources/migration-1.8.0_to_1.9.0/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom18to19.java

```diff
@@ -0,0 +1,702 @@
+     * This method is used to remove the FK constraint which is unnamed
+     * This finds the name of the constraint and build the query to delete the constaint and execute it
+     *
+     * @param migrateVersion version to be migrated
+     * @param dbType         database type of the user
+     * @throws SQLException
+     * @throws IOException
+     * @throws APIMigrationException
+     */
+    private void dropFKConstraint(String migrateVersion, String dbType) throws SQLException, IOException, APIMigrationException {
+        String constraintName = null;
+        Connection connection;
+        String queryToExecute = ResourceUtil.pickQueryFromResources(migrateVersion, Constants.CONSTRAINT).trim();
+        String queryArray[] = queryToExecute.split(Constants.LINE_BREAK);
+
+        connection = APIMgtDBUtil.getConnection();
+        connection.setAutoCommit(false);
+        Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery(queryArray[0]);
+        while (resultSet.next()) {
+            constraintName = resultSet.getString("constraint_name");
+        }
+
+        if (constraintName != null) {
+            queryToExecute = queryArray[1].replace("<temp_key_name>", constraintName);
+            PreparedStatement preparedStatement = connection.prepareStatement(queryToExecute);
+            preparedStatement.execute();
+            connection.commit();
+            preparedStatement.close();
+        }
+        connection.close();
+    }
+
+    /**
+     * This method is used to migrate all registry resources
+     * This migrates swagger resources and rxts
+     *
+     * @throws APIMigrationException
+     */
+    @Override
+    public void registryResourceMigration() throws APIMigrationException {
+        swaggerResourceMigration();
+        rxtMigration();
+    }
+
+    /**
+     * This method is used to migrate all the file system components
+     * such as sequences and synapse files
+     *
+     * @throws APIMigrationException
+     */
+    @Override
+    public void fileSystemMigration() throws APIMigrationException {
+        synapseAPIMigration();
+        sequenceMigration();
+    }
+
+    /**
+     * This method is used to migrate swagger v1.2 resources to swagger v2.0 resource
+     * This reads the swagger v1.2 doc from the registry and creates swagger v2.0 doc
+     *
+     * @throws APIMigrationException
+     */
+    void swaggerResourceMigration() throws APIMigrationException {
+        log.info("Swagger migration for API Manager 1.9.0 started.");
+        try {
+            for (Tenant tenant : tenantsArray) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Swagger migration for tenant " + tenant.getDomain() + "[" + tenant.getId() + "]" + " ");
+                }
+
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenant.getDomain());
```

Comment:
You can use PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenant.getDomain(), true) instead of setting the tenant id by hand.